src/integrationTest/java/com/hivemq/client/restrictions/Mqtt5ToMqtt3MessageForwardingIT.java

```diff
@@ -0,0 +1,221 @@
+
+public class Mqtt5ToMqtt3MessageForwardingIT {
+
+    @RegisterExtension
+    public final @NotNull HiveMQTestContainerExtension hivemq =
+            new HiveMQTestContainerExtension().withHiveMQConfig(MountableFile.forClasspathResource("/config.xml"));
+
+    @Test
+    void mqtt5ToMqtt3_messageForwarding_rx() throws InterruptedException {
+
+        final Mqtt3RxClient mqtt3Client = MqttClient
+                .builder()
+                .useMqttVersion3()
+                .serverPort(hivemq.getMqttPort())
+                .addConnectedListener(__ -> System.out.println("MQTTv3 client connected."))
+                .addDisconnectedListener(context -> System.out.println("MQTTv3 client disconnected. (" + context.getCause().getMessage() + ")"))
+                .buildRx();
+
+        final Mqtt5RxClient mqtt5Client = MqttClient
+                .builder()
+                .useMqttVersion5()
+                .serverPort(hivemq.getMqttPort())
+                .addConnectedListener(__ -> System.out.println("MQTTv5 client connected."))
+                .addDisconnectedListener(context -> System.out.println("MQTTv5 client disconnected. (" + context.getCause().getMessage() + ")"))
+                .buildRx();
+
+        final Mqtt3ConnAck mqtt3ConnAck = mqtt3Client.connect().timeout(5, TimeUnit.SECONDS).blockingGet();
+        assertEquals(Mqtt3ConnAckReturnCode.SUCCESS, mqtt3ConnAck.getReturnCode());
+
+        final Mqtt5ConnAck mqtt5ConnAck = mqtt5Client.connect().timeout(5, TimeUnit.SECONDS).blockingGet();
+        assertEquals(Mqtt5ConnAckReasonCode.SUCCESS, mqtt5ConnAck.getReasonCode());
+
+        final int MESSAGE_COUNT = 10;
+        final CountDownLatch countDownLatch = new CountDownLatch(MESSAGE_COUNT);
+
+        mqtt5Client
+                .publishes(MqttGlobalPublishFilter.ALL)
+                .subscribe(message -> {
+                    final byte[] payload = message.getPayloadAsBytes();
+                    final int i = payload[0];
+                    System.out.println("MQTTv5 client received message #" + i + " from topic \"test_1\".");
+                    System.out.println("MQTTv3 client sending message #" + i + " to topic \"test_2\"...");
+                    try {
+                        mqtt3Client
+                                .publish(
+                                        Flowable.just(
+                                                Mqtt3Publish
+                                                        .builder()
+                                                        .topic("test_2")
+                                                        .payload(payload)
+                                                        .qos(MqttQos.EXACTLY_ONCE)
+                                                        .build()
+                                        )
+                                )
+                                .timeout(5, TimeUnit.SECONDS)
+                                .subscribe();
```

Comment:
This code does not block the thread, it just starts publishing but does not wait for the acknowledgement.
If you would change this to a `blockingSubscribe`, you don't see the failure here as well.

Reply:
Your claim is _not_ true, though: the test still passes with `blockingSubscribe()`.

Reply:
ok, let me check if I also made a different change, because I tried this out before commenting.